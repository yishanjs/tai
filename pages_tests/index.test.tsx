/**
 * Copyright (c) 2019 Yishan Authors
 *
 * All right reserved
 */

import * as React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import Index, { Banner, Section } from '@/pages/index';

describe('首页', () => {
  it('snapshot 必须保持一致', async () => {
    const component = renderer.create(<Index />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('Banner ', () => {
    const banner = shallow(<Banner />);
    it('标题 必须为"确立目标 & 达成理想"', () => {
      expect(banner.find('h1').text()).toEqual('确立目标 & 达成理想');
    });
  });

  describe('Section ', () => {
    const baseProps = {
      title: 'test',
      description: ['我没说过'],
      credit: '鲁迅',
      imgSrc: '/public/image.png',
    };
    const section = mount(<Section {...baseProps} />);
    it('默认 dir 为 left，flexDirection 为 row', () => {
      expect(section.find('.content').prop('style').flexDirection).toEqual(
        'row'
      );
    });
    it('如果 dir 为 right，flexDirection 为 row-reverse', () => {
      section.setProps({ dir: 'right' });
      expect(section.find('.content').prop('style').flexDirection).toEqual(
        'row-reverse'
      );
    });
  });
});
