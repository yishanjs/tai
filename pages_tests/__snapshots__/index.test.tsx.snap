// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`首页 snapshot 必须保持一致 1`] = `
<div>
  <div
    className="bp3-navbar"
  >
    <div
      className="bp3-navbar-group bp3-align-left"
    >
      <div
        className="bp3-navbar-heading"
      >
        <img
          alt="logo"
          src="/static/layout/logo.png"
          style={
            Object {
              "height": 30,
              "width": 30,
            }
          }
        />
      </div>
      <div
        className="bp3-navbar-heading"
      >
        移山
      </div>
    </div>
  </div>
  <main
    className="jsx-4001454894"
  >
    <section
      className="jsx-4079040963"
    >
      <div
        className="jsx-4079040963 content"
      >
        <h1
          className="jsx-4079040963"
        >
          确立目标 & 达成理想
        </h1>
        <p
          className="jsx-4079040963"
        >
          不仅仅是代办事项
        </p>
        <a
          className="bp3-button bp3-intent-primary"
          href="https://github.com/login/oauth/authorize?client_id=undefined&redirect_uri=http://dev.yishan.co/oauth/github/redirect"
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            className="bp3-button-text"
          >
            内测用户登录
          </span>
        </a>
      </div>
      <img
        alt="people using yishan to get keep on their goal"
        className="jsx-4079040963"
        src="/images/banner.jpg"
      />
    </section>
    <section
      className="jsx-2872071450"
    >
      <div
        className="jsx-2872071450 content"
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <div
          className="jsx-2872071450 desc"
        >
          <h1
            className="bp3-heading"
            style={
              Object {
                "marginBottom": 20,
                "textAlign": "center",
              }
            }
          >
            设立长远目标
          </h1>
          <p
            className="jsx-2872071450 bp3-text-large"
          >
            所有的都将围绕长远价值展开。It's All About the Long Term.
          </p>
          <p
            className="jsx-2872071450 bp3-text-muted credit"
          >
            - 
            贝索斯1997年致股东的信
          </p>
        </div>
        <div
          className="jsx-2872071450 cover"
        >
          <img
            alt="设立长远目标"
            className="jsx-2872071450"
            src="/images/landing/long-term.jpeg"
          />
        </div>
      </div>
    </section>
    <section
      className="jsx-2872071450"
    >
      <div
        className="jsx-2872071450 content"
        style={
          Object {
            "flexDirection": "row-reverse",
          }
        }
      >
        <div
          className="jsx-2872071450 desc"
        >
          <h1
            className="bp3-heading"
            style={
              Object {
                "marginBottom": 20,
                "textAlign": "center",
              }
            }
          >
            反思与总结
          </h1>
          <p
            className="jsx-2872071450 bp3-text-large"
          >
            君子博学而日参省乎己，则知明而行无过矣。
          </p>
          <p
            className="jsx-2872071450 bp3-text-muted credit"
          >
            - 
            《荀子·劝学》
          </p>
        </div>
        <div
          className="jsx-2872071450 cover"
        >
          <img
            alt="反思与总结"
            className="jsx-2872071450"
            src="/images/landing/re-introspect.png"
          />
        </div>
      </div>
    </section>
    <section
      className="jsx-2872071450"
    >
      <div
        className="jsx-2872071450 content"
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <div
          className="jsx-2872071450 desc"
        >
          <h1
            className="bp3-heading"
            style={
              Object {
                "marginBottom": 20,
                "textAlign": "center",
              }
            }
          >
            记录你的一万小时
          </h1>
          <p
            className="jsx-2872071450 bp3-text-large"
          >
            1万小时的锤炼是任何人从平凡变成超凡的必要条件。
          </p>
          <p
            className="jsx-2872071450 bp3-text-muted credit"
          >
            - 
            格拉德威尔《异类》
          </p>
        </div>
        <div
          className="jsx-2872071450 cover"
        >
          <img
            alt="记录你的一万小时"
            className="jsx-2872071450"
            src="/images/landing/10000.jpg"
          />
        </div>
      </div>
    </section>
  </main>
  <div
    className="bp3-divider"
  />
  <footer
    style={
      Object {
        "color": "#8c8c8c",
        "margin": "10px",
        "textAlign": "center",
      }
    }
  >
    © 
    2019
     Yishan Authors
  </footer>
</div>
`;
